docs(ios-ui-tests): document terminate-after-navigation race + when to use direct tap vs helper

Following [#319](https://github.com/auerbachb/still-point/pull/319) (closes [#318](https://github.com/auerbachb/still-point/issues/318)) we landed a fix for the iOS smoke-lane flake in \`testLaunchLoginCompleteSessionAndHistoryPersistence\`. The investigation surfaced a non-obvious pitfall that is worth pinning to repo docs so future test authors do not re-discover it.

## The pitfall

When a UI test calls \`XCUIApplication.terminate()\` shortly after triggering navigation (e.g., immediately after a button \`.tap()\` that starts a new screen), \`launchd\` on macos-26/iOS 26 simulators sometimes loses the process and surfaces \`Failed to terminate ... :0\` after a long hang. Original failing-runs evidence: \`25215904325\`, \`25216409703\`, \`25216402153\`, \`25216777439\`.

## Why the obvious helper isn't always the right call

The natural reach is for the existing \`tap(_:thenWaitForRoot:in:)\` helper at [\`ios/StillPointAppUITests/StillPointAppUITests.swift:396\`](https://github.com/auerbachb/still-point/blob/main/ios/StillPointAppUITests/StillPointAppUITests.swift#L396). But that helper composes \`tapByStableCenter\` (8s stable-frame check) with a 12s root-existence wait, and \`tapByStableCenter\` returns \`false\` *without tapping* if the element's frame doesn't pass the stability check. In one specific flow (post-login → home → Begin) the home button frame failed that check; the helper retried 3× and never registered a tap. Evidence: PR #319 v1 (\`d9e69aa\`) failed CI on [run 25218186115](https://github.com/auerbachb/still-point/actions/runs/25218186115/job/73943514113) with \`StillPointAppUITests.swift:404: error: ... Expected tap to transition to root.currentView.session\`.

The shipping fix on \`main\` (\`ddf7b69\`) keeps direct \`beginButton.tap()\` and adds an explicit \`waitForExistence\` for the destination root before \`terminateAppReliably\` — same idle-before-terminate benefit, no change to tap semantics.

## What we want documented

Add a **\"Patterns and pitfalls\"** section to [\`ios/StillPointAppUITests/README.md\`](https://github.com/auerbachb/still-point/blob/main/ios/StillPointAppUITests/README.md) so future test authors have a clear rule. Concretely:

- One short entry: **\"Wait for the destination root before terminating after navigation.\"**
- Explain the launchd race (terminate during navigation/audio init).
- Show the \"good\" pattern (direct \`.tap()\` + \`waitForExistence(\"root.currentView.<slug>\")\` before \`terminateAppReliably\`).
- Note when \`tap(_:thenWaitForRoot:in:)\` is fine (already-quiescent flows: rotation/VoiceOver tests use it successfully) and when it can bite (post-login transitions where the destination element's frame is briefly unstable).
- Cite the actual file/line pointers and PR/run IDs so the rationale is auditable.

## Acceptance criteria

- [ ] New \`## Patterns and pitfalls\` section added to \`ios/StillPointAppUITests/README.md\`
- [ ] Section explains the terminate-after-navigation race in plain English
- [ ] Section gives the canonical \"good\" pattern with a Swift snippet
- [ ] Section explains when the \`tap(_:thenWaitForRoot:in:)\` helper is OK vs not
- [ ] Section links to PR #319 and the relevant failing/passing run IDs for traceability
- [ ] No code changes outside \`ios/StillPointAppUITests/README.md\`

## Implementation Plan

Source: Claude's analysis. CodeRabbit plan requested but doc-only changes typically don't need a planner round-trip; will incorporate any CR feedback during PR review.

### File
- `ios/StillPointAppUITests/README.md` (only)

### Insertion
A new `## Patterns and pitfalls` section, placed after `## CI setup de-duplication notes` and before `## VoiceOver smoke steps (manual fallback)`. ~50 lines.

### Section structure
1. Heading `## Patterns and pitfalls`
2. Subsection: **Wait for the destination root before terminating after navigation**
   - Plain-English explanation of the launchd race (failure signature `Failed to terminate ... :0`)
   - "Do this" Swift snippet using direct `.tap()` + `waitForExistence("root.currentView.<slug>")` + `terminateAppReliably`
   - "Don't do this" counter-example
3. Subsection: **When `tap(_:thenWaitForRoot:in:)` is fine vs when it bites**
   - Brief description of what the helper does (`tapByStableCenter` 8s stable-frame check + 12s root wait, 3× retry)
   - The trap: stable-frame check returns false silently when source element frame is briefly unstable (post-transition), helper retries 3× without ever tapping
   - Concrete evidence: PR [#319](https://github.com/auerbachb/still-point/pull/319) v1 commit `d9e69aa` failed [run 25218186115](https://github.com/auerbachb/still-point/actions/runs/25218186115); shipping fix `ddf7b69` keeps direct `.tap()`
   - Rule of thumb: helper for already-quiescent flows; direct `.tap()` + explicit wait right after a transition

### Out of scope
- No code changes to `StillPointAppUITests.swift` or any product code
- Not adding a lint rule / CI check (separate ticket if desired)
- Not updating `docs/testing/e2e-policy.md` — its Section 3 already mandates "XCTest expectations/predicate waits" at the policy level; this README is the how-to for iOS specifically

### Verification
- Local `coderabbit review --prompt-only` clean
- Open PR with Test Plan; CodeRabbit + BugBot review; merge after gate met (per repo policy)

--- a/ios/StillPointAppUITests/README.md
+++ b/ios/StillPointAppUITests/README.md
@@ -73,7 +73,7 @@ Planned lane: `ios-ui-smoke`
 
 If a test triggers navigation (taps a button that pushes a new screen, starts audio, kicks off any non-trivial transition) and then calls `app.terminate()`, the simulator's `launchd` on macos-26 / iOS 26 sometimes loses track of the process and hangs for tens of seconds before reporting:
 
-```
+```text
 Failed to terminate <bundle>:<pid>
 ```
 
@@ -105,7 +105,7 @@ The `tap(_:thenWaitForRoot:in:)` helper in [`StillPointAppUITests.swift`](StillP
 
 The trap: in flows where the source element's frame is briefly unstable (e.g. immediately after the home screen mounts following a fresh login), `tapByStableCenter` returns `false` *without ever tapping* — the helper then waits 12s for a root that never appears, retries, and ultimately asserts `Expected tap to transition to root.currentView.<slug>`.
 
-Concrete case: the v1 commit (`d9e69aa`) of [#319](https://github.com/auerbachb/still-point/pull/319) used the helper for the post-login `home → session` transition in `testLaunchLoginCompleteSessionAndHistoryPersistence` and failed CI on [run 25218186115](https://github.com/auerbachb/still-point/actions/runs/25218186115). The shipping fix (`ddf7b69`) kept the direct `.tap()` and added an explicit `waitForExistence` for the destination root.
+Concrete case: the v1 commit (`d9e69aa`) of [#319](https://github.com/auerbachb/still-point/pull/319) used the helper for the post-login `home → session` transition in `testLaunchLoginCompleteSessionAndHistoryPersistence` and failed CI on [run 25218186115](https://github.com/auerbachb/still-point/actions/runs/25218186115). The shipping fix (`ddf7b69`) kept the direct `.tap()` and added an explicit `waitForExistence` for the destination root — verified passing on [run 25218888601](https://github.com/auerbachb/still-point/actions/runs/25218888601).
 
 **Rule of thumb:**
 

--- a/ios/StillPointAppUITests/README.md
+++ b/ios/StillPointAppUITests/README.md
@@ -67,6 +67,51 @@ Planned lane: `ios-ui-smoke`
 - **Cross-reference #155/#156:** reuse shared simulator boot, xcodegen generation, and cache priming steps from those issues so iOS lanes do not duplicate setup work unnecessarily.
 - Keep auth/session fixture bootstrapping in app-side launch environment (this folder) rather than cloning separate backend setup in each lane.
 
+## Patterns and pitfalls
+
+### Wait for the destination root before terminating after navigation
+
+If a test triggers navigation (taps a button that pushes a new screen, starts audio, kicks off any non-trivial transition) and then calls `app.terminate()`, the simulator's `launchd` on macos-26 / iOS 26 sometimes loses track of the process and hangs for tens of seconds before reporting:
+
+```
+Failed to terminate <bundle>:<pid>
+```
+
+The fix is to wait for the destination's `root.currentView.<slug>` element to exist *before* terminating, so SIGTERM hits a quiescent process.
+
+**Do this:**
+
+```swift
+beginButton.tap()
+XCTAssertTrue(
+    app.otherElements["root.currentView.session"].waitForExistence(timeout: 20),
+    "Session screen did not appear after Begin tap"
+)
+app.launchEnvironment["SP_UI_TEST_SEED_AUTH"] = "1"
+terminateAppReliably(app)
+```
+
+**Don't do this:**
+
+```swift
+beginButton.tap()
+// terminate races launchd while audio engine + timer are still spinning up
+terminateAppReliably(app)
+```
+
+### When `tap(_:thenWaitForRoot:in:)` is fine vs when it bites
+
+The `tap(_:thenWaitForRoot:in:)` helper in [`StillPointAppUITests.swift`](StillPointAppUITests.swift) is a one-liner for "tap, then wait for the destination root". Tests like `testRotationDecisionSessionRemainsUsableInLandscape` use it successfully. Internally it composes `tapByStableCenter` (an 8s stable-frame check) with a 12s root-existence wait, retried up to 3 times.
+
+The trap: in flows where the source element's frame is briefly unstable (e.g. immediately after the home screen mounts following a fresh login), `tapByStableCenter` returns `false` *without ever tapping* — the helper then waits 12s for a root that never appears, retries, and ultimately asserts `Expected tap to transition to root.currentView.<slug>`.
+
+Concrete case: the v1 commit (`d9e69aa`) of [#319](https://github.com/auerbachb/still-point/pull/319) used the helper for the post-login `home → session` transition in `testLaunchLoginCompleteSessionAndHistoryPersistence` and failed CI on [run 25218186115](https://github.com/auerbachb/still-point/actions/runs/25218186115). The shipping fix (`ddf7b69`) kept the direct `.tap()` and added an explicit `waitForExistence` for the destination root.
+
+**Rule of thumb:**
+
+- **Already-quiescent flows** (authenticated boot, lighter session config, no fresh transition in flight): `tap(_:thenWaitForRoot:in:)` is fine.
+- **Right after a transition** (fresh login → home → tap, layout/animation likely still settling): use direct `.tap()` + explicit `waitForExistence` for the destination root.
+
 ## VoiceOver smoke steps (manual fallback)
 
 1. Enable VoiceOver in the simulator (`Settings -> Accessibility -> VoiceOver`).

